pkg/processor/trigger/kafka/trigger.go

```diff
@@ -120,8 +128,77 @@ func (k *kafka) GetConfig() map[string]interface{} {
 	return common.StructureToMap(k.configuration)
 }
 
-func (k *kafka) newKafkaConfig() (*sarama.Config, error) {
+func (k *kafka) Setup(session sarama.ConsumerGroupSession) error {
+	var err error
+
+	k.Logger.InfoWith("Starting consumer session",
+		"claims", session.Claims(),
+		"memberID", session.MemberID())
+
+	k.partitionWorkerAllocator, err = k.createPartitionWorkerAllocator(session)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create partition worker allocator")
+	}
+
+	return nil
+}
+
+func (k *kafka) Cleanup(session sarama.ConsumerGroupSession) error {
+	k.Logger.InfoWith("Ending consumer session",
```

Comment:
maybe release the allocated workers?

Reply:
good point - but i actually can't and MUST assume that I get here after sarama drained the messages (which would return all workers)